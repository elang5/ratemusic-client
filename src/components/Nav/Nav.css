#sidenav {
  height: 100%;
  position: fixed;
  width: 15%;
  background-color: #1f1f21;
  transition: 0.5s;
  display: flex;
  flex-flow: column nowrap;
}

.nav-item{
  width: 100%;
  text-align: center;
  height: 16.7%;
  cursor: pointer;
}

.link {
  text-decoration: none;
  color: white;
  transition: 0.3s;
}

div.logo {
  background-color: #1f1f21;
}

.nav-text {
  font-weight: bold;
  color: rgba(252, 61, 57, 0.73);
  margin: 0 auto;
  padding: 15px;
}

@media only screen and (max-width: 720px) {

  #sidenav {
    width: 28%;
  }

  .nav-text {
    max-width: 85%;
  }

  div.nav-item.social {
    display: flex;
    flex-flow: column nowrap;
  }

}

@media only screen and (max-width: 500px) {
  .nav-text {
    font-size: 14px;
  }
}

.svg.menu {
  width: 50px;
}

.svg {
  height: 20%;
  width: 50px;
  margin-top: 30px;
}

.nav-item:hover, .nav-item:active {
  background-color: rgba(252, 60, 57, 0.345);
}

div.albums {
  background-color: #1f1f21;
}

#closebtn, #openbtn {
  cursor: pointer;
}

div.nav-item.social {
  background-color: #1f1f21;
  display: flex;
  align-content: space-around;
  flex-flow: column nowrap;
  justify-content: space-around;
}

div.open {
  position: fixed;
  padding-left: 5px;
  height: 20%;
}

.svg-social {
  width: 50px;
  height: auto;
}

