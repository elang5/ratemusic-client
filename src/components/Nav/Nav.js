import React, { useState } from 'react'
import './Nav.css'
import Logo from '../../svg/Logo'
import Register from '../../svg/Register'
import Login from '../../svg/Login'
import Album from '../../svg/Album'
import Menu from '../../svg/Menu'
import LinkedIn from '../../svg/LinkedIn';
import Spotify from '../../svg/Spotify';
import { NavLink } from 'react-router-dom'

export default function Nav() {
  const [open, setOpen ] = useState(false)

  const handleSetOpen = (e) => {
    setOpen(true);
    e.currentTarget.style.visibility = "hidden";
  }

  const handleSetClose = (e) => {
    setOpen(false);
    const openMenu = document.getElementById('open-container');
    openMenu.style.visibility = "visible";
  }

  return (
    <>
      <nav 
        role="navigation" 
        className="navbar" 
        id="sidenav" 
        style={open ? {marginLeft: "0"} : {marginLeft: "-30.4%"}}>
        <div 
          className="nav-item close menu" 
          onClick={(e) => handleSetClose(e)} >
          <Menu 
            id="closebtn" 
            className="svg menu" 
            />
            <h3 className="nav-text">CLOSE</h3>
        </div>
        <NavLink 
          className="nav-item link logo" 
          to={"/albums"} 
          onClick={(e) => handleSetClose(e)}
        >
          <Logo className="svg logo"/>
          <h3 className="nav-text logo">RM</h3> 
        </NavLink>
        <NavLink 
          className="nav-item link register" 
          to={"/register"} 
          onClick={(e) => handleSetClose(e)}
        >
          <Register className="svg register" />
          <h3 className="nav-text register">REGISTER</h3>
        </NavLink>
        <NavLink 
          className="nav-item link albums" 
          to={"/albums"}
          onClick={(e) => handleSetClose(e)}
        >
          <Album className="svg album" />
          <h3 className="nav-text albums">ALBUMS</h3>
        </NavLink>
        <NavLink 
          className="nav-item link login" 
          to={"/login"}
          onClick={(e) => handleSetClose(e)}
        >
          <Login className="svg login" />
          <h3 className="nav-text login">LOGIN</h3>
        </NavLink>
        <div className="nav-item social">
          <a href="https://www.linkedin.com/in/elang589/">
            <LinkedIn className="svg-social linkedin" />
          </a>
          <a href='https://open.spotify.com/user/elang5'>
            <Spotify className="svg-social spotify" />
          </a>
        </div>
      </nav>
      <div 
        className="open"
        id="open-container"
        onClick={(e) => handleSetOpen(e)}
        >
        <Menu 
          id="openbtn" 
          className="svg menu closebtn" 
        />
      </div>
    </>
  )
}
