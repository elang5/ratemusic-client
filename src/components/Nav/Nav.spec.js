import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Nav from './Nav'

describe(`Nav Component`, () => {

  it(`renders the Nav component`, () => {
    const wrapper = shallow(<Nav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})