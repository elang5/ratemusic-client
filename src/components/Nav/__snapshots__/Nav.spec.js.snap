// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders the Nav component 1`] = `
<Fragment>
  <nav
    className="navbar"
    id="sidenav"
    role="navigation"
    style={
      Object {
        "marginLeft": "-30.4%",
      }
    }
  >
    <div
      className="nav-item close menu"
      onClick={[Function]}
    >
      <menuSVG
        className="svg menu"
        id="closebtn"
      />
      <h3
        className="nav-text"
      >
        CLOSE
      </h3>
    </div>
    <NavLink
      className="nav-item link logo"
      onClick={[Function]}
      to="/albums"
    >
      <logoSVG
        className="svg logo"
      />
      <h3
        className="nav-text logo"
      >
        RM
      </h3>
    </NavLink>
    <NavLink
      className="nav-item link register"
      onClick={[Function]}
      to="/register"
    >
      <registerSVG
        className="svg register"
      />
      <h3
        className="nav-text register"
      >
        REGISTER
      </h3>
    </NavLink>
    <NavLink
      className="nav-item link albums"
      onClick={[Function]}
      to="/albums"
    >
      <albumSVG
        className="svg album"
      />
      <h3
        className="nav-text albums"
      >
        ALBUMS
      </h3>
    </NavLink>
    <NavLink
      className="nav-item link login"
      onClick={[Function]}
      to="/login"
    >
      <loginSVG
        className="svg login"
      />
      <h3
        className="nav-text login"
      >
        LOGIN
      </h3>
    </NavLink>
    <div
      className="nav-item social"
    >
      <a
        href="https://www.linkedin.com/in/elang589/"
      >
        <linkedInSVG
          className="svg-social linkedin"
        />
      </a>
      <a
        href="https://open.spotify.com/user/elang5"
      >
        <spotifySVG
          className="svg-social spotify"
        />
      </a>
    </div>
  </nav>
  <div
    className="open"
    id="open-container"
    onClick={[Function]}
  >
    <menuSVG
      className="svg menu closebtn"
      id="openbtn"
    />
  </div>
</Fragment>
`;
